# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLContextGLX.h
# Commit: 16c5dd2cfd60
# Full Hash: 16c5dd2cfd600a62c80ca17cfb35c0868e016056
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r ca8139d58d58 -r 16c5dd2cfd60 gfx/gl/GLContextGLX.h
--- a/gfx/gl/GLContextGLX.h	Fri Jun 12 21:53:14 2020 +0000
+++ b/gfx/gl/GLContextGLX.h	Mon Jun 15 18:25:55 2020 +0000
@@ -20,9 +20,8 @@
  public:
   MOZ_DECLARE_REFCOUNTED_VIRTUAL_TYPENAME(GLContextGLX, override)
   static already_AddRefed<GLContextGLX> CreateGLContext(
-      CreateContextFlags flags, const SurfaceCaps& caps, bool isOffscreen,
-      Display* display, GLXDrawable drawable, GLXFBConfig cfg,
-      bool deleteDrawable, gfxXlibSurface* pixmap);
+      const GLContextDesc&, Display* display, GLXDrawable drawable,
+      GLXFBConfig cfg, bool deleteDrawable, gfxXlibSurface* pixmap);
 
   static bool FindVisual(Display* display, int screen, bool useWebRender,
                          bool useAlpha, int* const out_visualId);
@@ -66,8 +65,7 @@
  private:
   friend class GLContextProviderGLX;
 
-  GLContextGLX(CreateContextFlags flags, const SurfaceCaps& caps,
-               bool isOffscreen, Display* aDisplay, GLXDrawable aDrawable,
+  GLContextGLX(const GLContextDesc&, Display* aDisplay, GLXDrawable aDrawable,
                GLXContext aContext, bool aDeleteDrawable, bool aDoubleBuffered,
                gfxXlibSurface* aPixmap);
 